AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java

```diff
@@ -2738,6 +2742,18 @@ public void actualOnPreExecute(@NonNull DeckPicker deckPicker) {
                     deckPicker.getResources().getString(R.string.emtpy_cards_finding), false);
         }
 
+        @Override
+        public void actualOnProgressUpdate(@NonNull DeckPicker deckPicker, @NonNull TaskData progress) {
+            int lastCardSeenPercent = (int) ((cardSeen * 100) / numberCards);
+            cardSeen += progress.getInt();
+            int cardSeenPercent = (int) ((cardSeen * 100) / numberCards);
+            if (lastCardSeenPercent != cardSeenPercent) {
+                // Ensure we update at most 100 times. Updating in real time freeze the screen
+                deckPicker.mProgressDialog.setContent(
+                        deckPicker.getResources().getString(R.string.emtpy_cards_finding) + cardSeen + " / " + numberCards);
```

Comment:
```suggestion
                        deckPicker.getResources().getString(R.string.empty_cards_finding) + cardSeen + " / " + numberCards);
```

But more problematically this does not work with RTL languages, the numbers should be interpolated into the string with a string format I think?

And a minor note, but if the math is "fuzzy" (you mention that things might go up or down) then I think you could put a `~` sign or something before the numbers to indicate "not exact but close"

Reply:
I copied what we do for "check collection". You may be right, but in this case it means this is also incorrect.

The number shown are exact. The trouble is that both numbers counts different things.  
I.e. "39691 /233434" means "We have checked that 39691 cards should be generated from your notes according to your collection, and for information your collection has 233434 cards". 
This is a good indication, because assuming that there are only a small number of cards to create/delete, then the numerator will almost reach the denominator.
So "~" seems really strange here.
If instead I showed percent, that would makes sens. But I like the idea to showing to the user their actual number of cards

Reply:
Why not use a progress bar capped at 100%? 

Intuition from reading the code a while back implies that our Progress Dialog supports it, and this removes the ambiguity from a user's point of view - a progress bar stuck at 100% isn't great, but it happens a lot in applications, and a user would expect it.

Reply:
I tried to copy what I currently have. I like seeing the actual number of cards in the collection, I find it nice. But if you prefer any other UI, I'm fine with them too as long as it's not harder to code.

Reply:
Inexact progress bars drive me crazy personally (like, there is a numerator and a denominator but it is not really a fraction of work effort being represented...), and it seems that if we want to expose the actual counts we have all sorts of complexity with RTL and strings and stuff.

Let's just do a percent 0 to 100 capped at 100, life will be easy :-)

Reply:
I added a number with percent symbol, and a new string... not ideal because this new string contains a part of the old string. But let's go for it if that's what you prefer. 
I have little interest in UI I fear, and will do whatever I have to in order to get the feature I care about in the codebase. But I'll leave the beautiful UI to other if that's what they want to spend their time on.